src/components/ProductsList/ProductsList.tsx

```diff
@@ -1,35 +1,117 @@
+        setCurrentPage(1);
+    }, [filteredProductsData]);
+
+    const qtyOfPages = Math.ceil(filteredProductsData.length / productCardsOnPage);
+
+    function createButtons() {
+        const buttons = [];
+        for (let index = 1; index <= qtyOfPages; index++) {
+            const button = (
+                <button
+                    className={`${styles.pageButton} ${currentPage === index ? styles.active : ''}`}
+                    key={index}
+                    onClick={() => setCurrentPage(index)}
+                >
+                    {index}
+                </button>
+            );
+            buttons.push(button);
+        }
+        return buttons;
+    }
+
+    const pageButtons = createButtons();
+
+    function createPageButtons() {
+        let test;
+        if (currentPage <= 3) {
+            test = [
+                pageButtons[currentPage - 3],
+                pageButtons[currentPage - 2],
+                pageButtons[currentPage - 1],
+                pageButtons[currentPage],
+                <span key={Math.random()}>...</span>,
+                pageButtons[qtyOfPages - 1],
+            ];
+        }
+
+        if (currentPage >= qtyOfPages - 2) {
+            test = [
+                pageButtons[0],
+                <span key={Math.random()}>...</span>,
+                pageButtons[currentPage - 2],
+                pageButtons[currentPage - 1],
+                pageButtons[currentPage],
+                pageButtons[currentPage + 1],
+            ];
+        }
+
+        if (currentPage > 3 && currentPage < qtyOfPages - 2) {
+            test = [
+                pageButtons[0],
+                <span key={Math.random()}>...</span>,
+                pageButtons[currentPage - 2],
+                pageButtons[currentPage - 1],
+                pageButtons[currentPage],
+                <span key={Math.random()}>...</span>,
+                pageButtons[qtyOfPages - 1],
+            ];
+        }
+
+        return test;
+    }
 
-function ProductsList({ data, cartData, handleAddToCart }: ProductsListProps) {
     return (
         <>
             <SearchBar />
             <section className={styles.productsList}>
                 <div className="container">
                     <div className={styles.wrapper}>
-                        {data.map((productData) => (
-                            <ProductCard
-                                key={productData.title}
-                                productData={productData}
-                                onAddToCart={handleAddToCart}
-                                cartData={cartData}
-                            />
-                        ))}
+                        {isLoadingProductsData && <p>Loading Products...</p>}
+                        {isErrorFetchingProductsData && <p>Ops... Some Problem</p>}
+                        {filteredProductsData.length === 0 && !isErrorFetchingProductsData && !isLoadingProductsData && <p>No Items</p>}
```

Comment:
```suggestion
const isNoItems = filteredProductsData.length === 0 && !isErrorFetchingProductsData && !isLoadingProductsData;
                        {isNoItems && <p>No Items</p>}
```

if u have more than one logical condition it is worth to move them into separate variable, it will help to make your JSX more readable